libs/platform/src/lib/components/form/select/commons/base-select.ts

```diff
@@ -181,8 +180,14 @@ export abstract class BaseSelect extends CollectionBaseInput implements AfterVie
      */
     @Input()
     set contentDensity(contentDensity: ContentDensity) {
-        this._contentDensity = contentDensity;
-        this.isCompact = contentDensity === 'compact';
+        if (contentDensity === undefined && this._contentDensityService) {
+            this._subscriptions.add(this._contentDensityService._contentDensityListener.subscribe(density => {
```

Comment:
every time when we set new `contentDensity` value new subscription will be created (even in case if we already might have one set previously)